### DISCUSSION TITLE
Material id node

### DISCUSSION THREAD

**Discussion by dibli-goost** [bnpr/Malt#321] [Feedback & Feature Requests]
This is partly related to this bug:  https://github.com/bnpr/Malt/issues/319

A big reason why this problem with the outline on the Eyes is happening is because without setting a custom id for _every_ material, Malt will inevitably run into a situation where it sees two materials as essentially the same for the sake of rendering lines. I've found another thread where this was brought up: https://github.com/bnpr/Malt/discussions/105

I want to reinforce the idea that we need this feature. Otherwise, we'll have to constantly deal with creating custom ids for materials when otherwise we should have it automatically generated.

In a similar vein. I really think that Malt should by _default_ use a combination of material ids and object ids when generating lines. But that's just me. Every separate object, and every separate material should be treated as intrinsically different from one another. I feel this would be a much better workflow for characters overall, where the largest amount of line art finagling occurs I think. 

From what I said on the discord: _"Personally, I don't really see the benefit in having Malt operate in this manner. Wouldn't it make more sense for it to by default see objects/individual materials as distinguished from one another? To me it feels like if we want Malt to look at two things as if the are the same, then we should be able to opt into that, shouldn't be the default. Wanting things to be seen as the same thing seems like an edge case is what I'm getting at. I'm sure there are cases where we might want that functionality, but it's not gonna be what we typically want from the program"_

**Comment by pragma37** [comment]
Whether to draw lines at material boundaries or not is not something the engine should force on the user.
The point of having 3 unused ID channels is to allow users to set them however they want, so in your case, you can use the second channel for material ids.
